app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,79 @@
+package com.codesoom.assignment.application;
+
+import static org.assertj.core.api.Assertions.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+
+class TaskServiceTest {
+
+    private TaskService taskService;
+    private static final String TASK_TITLE = "test";
+    private static final String UPDATE_POSTFIX = "!!!";
+
+    @BeforeEach
+    void setUp() {
+        // subject
+        taskService = new TaskService();
+
+        // fixture
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        taskService.createTask(task);
+    }
+
+    @Test
+    void getTasks() {
+        assertThat(taskService.getTasks()).hasSize(1);
+
+        Task task = taskService.getTasks().get(0);
+        assertThat(task.getTitle()).isEqualTo(TASK_TITLE);
+    }
+
+    @Test
+    void getTaskWithValidId() {
+        Task found = taskService.getTask(1L);
+        assertThat(found.getTitle()).isEqualTo(TASK_TITLE);
+    }
+
+    @Test
+    void getTaskWithInvalidId() {
+        assertThatThrownBy(() -> taskService.getTask(100L))
+            .isInstanceOf(TaskNotFoundException.class);
+    }
+
+    @Test
+    void createTask() {
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        taskService.createTask(task);
+        assertThat(taskService.getTasks()).hasSize(2);
+    }
```

Comment:
task를 만들었더니 왜 사이즈가 `2`일까요? 테스트는 일종의 우리 코드의 예제코드에요. 우리의 코드를 사용하는 사용자에게 우리의 코드를 사용하면 무엇을 할 수 있는지 나타내야 해요.
우리의 코드를 실행하면 무엇을 얻을 수 있고 어떤 `변화`가 일어나는지 주목해야 해요. 
아래의 테스트는 어떤 변화가 일어나는지를 의도를 드러내고 있어요.
보니까 아래의 다른 코드에서는 아주 잘 해주셨군요 ㅎㅎ
```suggestion
    @Test
    void createTask() {
        Task task = new Task();
        task.setTitle(TASK_TITLE);
        Int size = taskService.getTasks();
        
        taskService.createTask(task);
        
        assertThat(taskService.getTasks().size() - size).isEqualTo(1);
    }
```

Reply:
정말로 그렇네요, 제가 작성한 코드지만 이 코드가 무엇을 나타내려고 하는지 이해할 수가 없네요 ㅋㅋㅋ